contracts/src/v0.8/functions/dev/1_0_0/FunctionsSubscriptions.sol

```diff
@@ -410,33 +423,54 @@ abstract contract FunctionsSubscriptions is IFunctionsSubscriptions, IERC677Rece
     emit SubscriptionConsumerAdded(subscriptionId, consumer);
   }
 
-  // @inheritdoc IFunctionsSubscriptions
-  function cancelSubscription(uint64 subscriptionId, address to) external override {
-    _whenNotPaused();
-    _onlySubscriptionOwner(subscriptionId);
-    _onlySenderThatAcceptedToS();
-
-    if (pendingRequestExists(subscriptionId)) {
-      revert CannotRemoveWithPendingRequests();
-    }
-
-    _cancelSubscriptionHelper(subscriptionId, to);
-  }
+  // Overriden in FunctionsRouter.sol
+  function _getSubscriptionDepositDetails() internal virtual returns (uint16, uint72);
 
-  function _cancelSubscriptionHelper(uint64 subscriptionId, address to) private {
+  function _cancelSubscriptionHelper(uint64 subscriptionId, address toAddress, bool checkDepositRefundability) private {
     Subscription memory subscription = s_subscriptions[subscriptionId];
     uint96 balance = subscription.balance;
+    uint64 completedRequests = 0;
+
     // NOTE: loop iterations are bounded by config.maxConsumers
     // If no consumers, does nothing.
     for (uint256 i = 0; i < subscription.consumers.length; ++i) {
-      delete s_consumers[subscription.consumers[i]][subscriptionId];
+      address consumer = subscription.consumers[i];
+      completedRequests += s_consumers[consumer][subscriptionId].completedRequests;
+      delete s_consumers[consumer][subscriptionId];
     }
     delete s_subscriptions[subscriptionId];
-    s_totalLinkBalance -= balance;
 
-    IERC20(i_linkToken).safeTransfer(to, uint256(balance));
+    (uint16 subscriptionDepositMinimumRequests, uint72 subscriptionDepositJuels) = _getSubscriptionDepositDetails();
+
+    IERC20 linkToken = IERC20(i_linkToken);
+
+    // If subscription has not made enough requests, deposit will be forfeited
+    if (completedRequests < subscriptionDepositMinimumRequests && !checkDepositRefundability) {
```

Comment:
With the new name "checkDepositRefundability", we want it to have an opposite effect , right? We run this if it's set to true.

Reply:
errr, yeah true